src/components/NavLink.tsx

```diff
@@ -24,19 +24,33 @@ export const NavLink = forwardRef<HTMLAnchorElement, NavLinkProps>(
     const isActive = pathname === href
     const isActiveParent = href !== '/' && pathname.startsWith(`${href}`)
 
+    const innerLinkContent = (
+      <>
+        {children}
+        {isExternal && !hideIcon ? (
+          <ExternalLinkIcon className="size-3.5 text-grayAlpha-400" />
+        ) : null}
+      </>
+    )
+
+    const linkContent = isExternal ? (
+      <a ref={ref} target={target} href={href}>
+        {innerLinkContent}
+      </a>
+    ) : (
+      <Link ref={ref} target={target} href={href}>
+        {innerLinkContent}
+      </Link>
+    )
```

Comment:
The Link component can be used with external links https://github.com/vercel/next.js/discussions/37597